src/util/firebase/index.ts

```diff
@@ -34,9 +35,12 @@ const auth = getAuth(app);
 
 const logIn = () => {
   signIn(auth, googleProvider, db);
+
   const upi = auth.currentUser?.email?.split('@')?.[0];
+  const userGroup = new Prando(upi).nextBoolean() ? 'A' : 'B';
```

Comment:
Should we be doing this on all log-ins or just the first? I'd assume it would be ok either way because Prando ensures the same result for the same UPI? Just want to confirm

Reply:
All logins to ensure that this is registered when the user logs in on a new device, browser etc.

Reply:
And yes, prando ensures the same result.